lib/e_pub/publication.rb

```diff
@@ -89,10 +90,14 @@ def purge
     end
 
     def read(file_entry = "META-INF/container.xml")
-      return Publication.null_object.read(file_entry) unless Cache.cached?(id)
-      entry_file = ::EPub.path_entry(id, file_entry)
+      if @root_path.present?
+        entry_file = File.join(root_path, file_entry)
+      else
+        return Publication.null_object.read(file_entry) unless Cache.cached?(id)
+        entry_file = ::EPub.path_entry(id, file_entry)
+      end
       return Publication.null_object.read(file_entry) unless File.exist?(entry_file)
-      FileUtils.touch(::EPub.path(id)) # Reset the time to live for the entire cached EPUB
+      FileUtils.touch(::EPub.path(id)) # if root_path.blank? # Reset the time to live for the entire cached EPUB
```

Comment:
Everything seems to be working smoothly but, even when unpacked the "new" way, this touch of the old path is happening, producing an empty file. Is that intentional, perhaps to continue leaving a record of all epubs in the cache dir? I see there is a commented conditional here (that would touch the new way only).

Reply:
Oops. Yes, thanks for catching this! The conditional should not be commented out (I was debugging and didn't uncomment it.). I'll fix it.